File: paint/sources/brush_nodes/null_node.ts

type null_node_t = {
	base?: logic_node_t;
};

function null_node_create(raw: ui_node_t, args: f32_array_t): null_node_t {
	let n: null_node_t = {};
	n.base = logic_node_create(n);
	n.base.get = float_node_get;
	return n;
}

function null_node_get(self: null_node_t, from: i32): logic_node_value_t {
	return null;
}

File: paint/sources/brush_nodes/null_node.c

#include "../global.h"

logic_node_value_t *null_node_get(null_node_t *self, i32 from) {
	return NULL;
}

null_node_t *null_node_create(ui_node_t *raw, f32_array_t *args) {
	null_node_t *n = GC_ALLOC_INIT(null_node_t, {0});
	n->base        = logic_node_create(n);
	n->base->get   = float_node_get;
	return n;
}
